gravitee-apim-portal-webui-next/src/app/registration/registration.component.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (C) 2025 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, DestroyRef, OnInit, signal } from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { FormControl, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
+import { MatButtonModule } from '@angular/material/button';
+import { MatCard, MatCardContent, MatCardHeader, MatCardTitle } from '@angular/material/card';
+import { MatIcon } from '@angular/material/icon';
+import { MatError, MatFormField, MatInput, MatLabel } from '@angular/material/input';
+import { MatOption, MatSelect } from '@angular/material/select';
+import { RouterLink } from '@angular/router';
+import { tap } from 'rxjs';
+
+import { MobileClassDirective } from '../../directives/mobile-class.directive';
+import { CustomUserFields } from '../../entities/user/custom-user-fields';
+import { UsersService } from '../../services/users.service';
+
+interface UserRegistrationFormValue {
+  firstname: string;
+  lastname: string;
+  email: string;
+  customFields?: { [key: string]: string };
+}
+
+@Component({
+  selector: 'app-registration',
+  imports: [
+    MatCard,
+    MatCardContent,
+    MatCardHeader,
+    MatCardTitle,
+    MatError,
+    MatFormField,
+    MatInput,
+    MatLabel,
+    MatButtonModule,
+    ReactiveFormsModule,
+    RouterLink,
+    MobileClassDirective,
+    MatError,
+    MatFormField,
+    MatOption,
+    MatSelect,
+    MatIcon,
+  ],
+  templateUrl: './registration.component.html',
+  styleUrl: './registration.component.scss',
+})
+export class RegistrationComponent implements OnInit {
+  submitted: boolean = false;
+  sentToEmail = '';
+
+  customUserFields: CustomUserFields[] = [];
```

Comment:
In order to ensure compatibility with future versions of Angular without Zone.js, we should use signals for all component properties that are reactive (= which trigger the template update).

For example, in the zoneless mode, when the usersService.registerNewUser() resolves and `submitted` is reassigned to true, the template will still show the page "Create an account".

```suggestion
submitted = signal(false);
sentToEmail = signal('');

customUserFields = signal<CustomUserFields[]>([]);